### ISSUE TITLE
editor:delete-line should preserve line position

### ISSUE THREAD

**Issue by rumkin** [atom/atom#12744]
### Description

The command `editor:delete-line` should preserve current cursor position after removing line.
### Steps to Reproduce

For example we have file `data.js`:

``` javascript
var arr = [
        'main',
        'user',
        'settings',
        'lang',
        'billing',
];
```
1. Put cursor in file at line `settings` after first quote.
2. Call delete-line command

**Expected behavior:** Cursor is at the `lang` line after first quote. Line position is 9.

**Actual behavior:** Cursor is at the start of  `lang` line. Line position is 0.
### Versions

Atom 1.10.2, macOS 10.10

**Comment by 50Wliu** [comment]
In the future, please **completely** fill out the provided issue template.  For starters, you're using an extremely outdated version of Atom: the latest version is currently Atom 1.10.2, which you would have realized if you went through the prerequisites.

In addition, I cannot reproduce this behavior on Atom 1.12.0-dev-c852317, Windows 10 x64, safe mode.  Following your reproduction steps, the cursor ends up at the start of the `];` line, which to me is the expected behavior.  Can you try reproducing this in safe mode (`atom --safe`)?  (Again, that was one of the prerequisites in the issue template.)

**Comment by rumkin** [comment]
@50Wliu I've updated atom. Add cursor position to clearify the issue.

I called `atom --safe` and issue is reproduced again.

**Comment by vedharish** [comment]
I can reproduce this (Started with --safe)
Atom    : 1.12.0-dev-9977aa5
Electron: 1.3.6
Chrome  : 52.0.2743.82
Node    : 6.3.0
MacOS : 10.12

![atomdeletelinecast](https://cloud.githubusercontent.com/assets/2497481/18771150/371ea0c8-815a-11e6-95f6-77e3fac21fd5.gif)

**Comment by 50Wliu** [comment]
Thanks for the extra reproduction steps @rumkin and for the gif @vedharish.  Reproduced.

**Comment by iolsen** [comment]
👍 this would be an improvement. We'd accept a well-tested PR that made this change.

**Comment by andzaytsev** [comment]
It seems like either of #12783 and #12786 should fix this issue. Is there something that should be changed in those PRs? If so, I could submit a new one.

**Comment by 50Wliu** [comment]
Fixed by #12783.
